__tests__/pathfinding.service.test.js

```diff
@@ -132,11 +112,80 @@ describe("Find the shortest path between two locations/POI's", () => {
     const path = findPathBetweenPOIs(
       POIInfo.find(
         ({ buildingId, displayName, level }) =>
-          displayName === "Women's Bathroom" && level === 8
+          displayName === "Women's Bathroom" &&
+          level === 8 &&
+          buildingId === "H"
       ),
-      POIInfo.find(({ displayName }) => displayName === "H961-17")
+      POIInfo.find(({ displayName }) => displayName === "H961-17"),
+      false
     );
     expect(path.length).toBe(2);
     expect(path).toEqual(testData.h8WomensBathroomToH9117);
   });
+
+  it("should return null when no path is found", () => {
+    // store the initial nodes
+    const initialNodes = _.cloneDeep(
+      constants.buildingFloors.find(
+        (floor) => floor.buildingId === "H" && floor.level === 9
+      ).travelNodes
+    );
+    constants.buildingFloors.find(
+      (floor) => floor.buildingId === "H" && floor.level === 9
+    ).travelNodes = testData.brokenGraph;
+    const shortest = findPathOnFloor(
+      POIInfo.find(({ displayName }) => displayName === "H961-19"),
+      POIInfo.find(({ displayName }) => displayName === "H911"),
+      false
+    );
+
+    expect(shortest).toBeNull();
+    // reset the initial nodes
+    constants.buildingFloors.find(
+      (floor) => floor.buildingId === "H" && floor.level === 9
+    ).travelNodes = initialNodes;
+  });
```

Comment:
Can you explain what is going on here and why this is needed at the end of the test?

Reply:
In line 133 to 135, I changed the travelNodes of H9 of constants.buildingFloors  to a broken graph. I need to change it back so the tests after that use constants.buildingFloors will not fail.

Reply:
But on line 128 you make a deepclone and put it in `initialNodes` why don't you do the test with the deepclone you made so you don't have to do all of 143-146.

Reply:
The function does not accept travelNodes as a parameter, it takes data from constants.buildingFloors directly, that's why I have to "mock" constants.buildingFloors. However, I could not find a better way to mock except temporarily changing constants.buildingFloors.

Reply:
I see, it's modified directly inside the function. Thanks for the info, I agree this is the only way at the moment.